tests/unit/s2n_release_non_empty_buffers_test.c

```diff
@@ -137,17 +138,18 @@ int main(int argc, char **argv)
 
         /* check to see if we need to copy more over from the pipes to the buffers
          * to continue the handshake */
-        s2n_stuffer_recv_from_fd(&in, io_pair.server, MAX_BUF_SIZE);
-        s2n_stuffer_send_to_fd(&out, io_pair.server, s2n_stuffer_data_available(&out));
+        GUARD(s2n_stuffer_recv_from_fd(&in, io_pair.server, MAX_BUF_SIZE, NULL));
+        GUARD(s2n_stuffer_send_to_fd(&out, io_pair.server, s2n_stuffer_data_available(&out), NULL));
     } while (blocked);
 
     /* Receive only 100 bytes of the record and try to call s2n_recv */
-    n = 0;
     while (n < 100) {
-        ret = s2n_stuffer_recv_from_fd(&in, io_pair.server, 100 - n);
+        ret = s2n_stuffer_recv_from_fd(&in, io_pair.server, 100 - n, &n);
 
-        if (ret < 0 && (errno == EAGAIN || errno == EWOULDBLOCK)) {
+        if (errno == EAGAIN || errno == EWOULDBLOCK) {
             continue;
+        } else {
+            GUARD(ret);
         }
 
         EXPECT_TRUE(ret > 0);
```

Comment:
This and the next line need to be removed since we're now double checking and adding to n

Reply:
Done.